msal_extensions/cache_lock.py

```diff
@@ -12,17 +12,22 @@ class CrossPlatLock(object):
     """
     def __init__(self, lockfile_path):
         self._lockpath = lockfile_path
-        self._fh = None
+        self._lock = portalocker.Lock(
+            lockfile_path,
+            mode='wb+',
+            # In posix systems, we HAVE to use LOCK_EX(exclusive lock) bitwise ORed
+            # with LOCK_NB(non-blocking) to avoid blocking on lock acquisition.
+            # More information here: https://docs.python.org/3/library/fcntl.html#fcntl.lockf
```

Comment:
Nit: Given that a URL tends to be long, it is usually better to put it in its own line to help keep lines within the 80-character long recommendation by PEP 8. I'm still going to give an approval on this PR. :-)

```suggestion
            # More information here:
            # https://docs.python.org/3/library/fcntl.html#fcntl.lockf
```